packages/tuono/src/ssr/server.tsx

```diff
@@ -1,9 +1,30 @@
-import 'fast-text-encoding' // Mandatory for React18
+import 'fast-text-encoding' // Mandatory TextEncoder and TextDecoder polyfills
+// eslint-disable-next-line import/order
+import { MessageChannelPolyfill } from './polyfills/MessageChannel'
+
+/**
+ * The polyfill for MessageChannel need to be done here.
+ * React might throw an error since it needs it right away,
+ * unlike other polyfills like ReadableStream or TextDecoder,
+ * and the polyfill itself is not hoisted
+ */
+// eslint-disable-next-line import/newline-after-import
+;(function (
+  scope: Partial<Pick<typeof globalThis, 'MessageChannel'>> = {},
+): void {
+  scope['MessageChannel'] = scope['MessageChannel'] ?? MessageChannelPolyfill
+})(
+  typeof window !== 'undefined'
+    ? window
+    : typeof global !== 'undefined'
+      ? global
+      : this,
+)
```

Comment:
I added an explanation to this workaround to provide `MessageChannel` polyfill.
Since it is a little bit "hacky" I would really consider to log a task to find a better alternative.

Reply:
#266